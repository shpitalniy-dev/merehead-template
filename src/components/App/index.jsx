import React, { Suspense } from 'react'
import Alert from 'react-s-alert';
import { Switch, Route } from 'react-router-dom';
import { useSelector } from 'react-redux'
import Modal from '../Modal';
import AlertTemplate from '../../libs/AlertTemplate';
import { Loader } from '../../libs/Loader';
import routes from '../../routes';
import AuthWrapper from '../../libs/AuthWrapper';
import NotAuthWrap from '../../libs/NotAuthWrap'
import { getAuthIsLoading } from '../../redux/Auth/Auth.selectors';

const App = () => {
  const isLoading = useSelector(getAuthIsLoading);

  return (
    <>
      {isLoading ? <Loader /> :
        <>
          <div className="notifications">
            <Alert
              contentTemplate={AlertTemplate}
              stack={{ limit: 2 }}
              effect="scale"
              html={true}
              position="top-right"
              timeout={3000}
              offset={50}
            />
          </div>
          <Suspense fallback={<Loader />}>
            <Modal />
            <Switch>
              <NotAuthWrap path={routes.Root.path} component={routes.Root.component} exact />
            </Switch>
          </Suspense>
        </>}
    </>
  );
};

export default App;
