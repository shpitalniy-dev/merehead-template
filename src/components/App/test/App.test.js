import App from '../index';

describe(`App component structure`, () => {
    it(`App component create a snapshot, should render correctly`, () => {
        const wrapper = shallowSmart(App, {});
        expect(wrapper).toMatchSnapshot();
    });
});