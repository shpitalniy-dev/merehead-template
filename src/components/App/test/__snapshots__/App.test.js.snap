// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component structure App component create a snapshot, should render correctly 1`] = `
<div
  className="page-loader"
  style={
    Object {
      "zIndex": 999,
    }
  }
>
  <div
    className="loader"
  />
</div>
`;
